application/src/test/java/org/thingsboard/server/controller/AlarmControllerTest.java

```diff
@@ -796,4 +800,101 @@ public void testCreateAlarmWithOtherCustomerAsAssignee() throws Exception {
         doPost("/api/alarm", alarm).andExpect(status().isForbidden());
     }
 
+    @Test
+    public void testGetAlarmTypes() throws Exception {
+        loginTenantAdmin();
+
+        List<String> types = new ArrayList<>();
+
+        for (int i = 1; i < 13; i++) {
+            types.add(createAlarm(TEST_ALARM_TYPE + i).getType());
+        }
+
+        List<String> foundTypes = doGetTyped("/api/alarm/types?pageSize=1024&page=0", new TypeReference<PageData<EntitySubtype>>() {
+        })
+                .getData()
+                .stream()
+                .map(EntitySubtype::getType)
+                .collect(Collectors.toList());
+
+        Collections.sort(types);
+        Collections.sort(foundTypes);
+
+        Assert.assertEquals(types, foundTypes);
+    }
+
+    @Test
+    public void testDeleteAlarmTypes() throws Exception {
+        loginTenantAdmin();
+
+        List<AlarmInfo> alarms = new ArrayList<>();
+
+        for (int i = 1; i < 13; i++) {
+            alarms.add(createAlarm(TEST_ALARM_TYPE + i));
+        }
+
+        Device device = new Device();
+        device.setName("Test device 2");
+        device.setCustomerId(customerId);
+        customerDevice = doPost("/api/device", device, Device.class);
+
+        for (int i = 1; i < 14; i++) {
+            alarms.add(createAlarm(TEST_ALARM_TYPE + i));
+        }
+
+        List<String> expectedTypes = alarms.stream().map(AlarmInfo::getType).distinct().sorted().collect(Collectors.toList());
+
+        List<String> foundTypes = doGetTyped("/api/alarm/types?pageSize=1024&page=0", new TypeReference<PageData<EntitySubtype>>() {
+        })
+                .getData()
+                .stream()
+                .map(EntitySubtype::getType)
+                .sorted()
+                .collect(Collectors.toList());
+
+        Assert.assertEquals(13, foundTypes.size());
+        Assert.assertEquals(expectedTypes, foundTypes);
+
+        for (int i = 0; i < 12; i++) {
+            doDelete("/api/alarm/" + alarms.get(i).getId()).andExpect(status().isOk());
+        }
+
+        foundTypes = doGetTyped("/api/alarm/types?pageSize=1024&page=0", new TypeReference<PageData<EntitySubtype>>() {
```

Comment:
Page size is better to get from the yaml value for default page size. if someone set the page size is too low, the test fill fail. 
See an example:
```
    @Value("${sql.ttl.audit_logs.ttl}")
    private long auditLogsTtlInSec;
```